### PR TITLE
 vscode-extensions.ms-python.flake8: init at 2023.6.0

### PR DESCRIPTION
**Pull request by carlthome**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -2851,6 +2851,22 @@ let
...
       };
 
+      ms-python.flake8 = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "flake8";
+          publisher = "ms-python";
+          version = "2023.6.0";
+          hash = "sha256-Hk7rioPvrxV0zMbwdighBAlGZ43rN4DLztTyiHqO6o4";
+        };
+        meta = with lib; {
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:2861]:
```suggestion
        meta = {
```
Avoid nested `with` whenever possible.

Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/default.nix:2861]: Consider reading the conventions for VSCode extensions.
Relevant points:
- changelog
- description
- [nixfmt-rfc-style](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=nixfmt-rfc-style)
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/vscode/extensions/README.md

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -1822,7 +1822,7 @@ let
         mktplcRef = {
           publisher = "github";
           name = "copilot-chat";
-          version = "0.14.2024032901";  # compatible with vscode 1.88.1
+          version = "0.14.2024032901"; # compatible with vscode 1.88.1
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:1825]:
```suggestion
          version = "0.14.2024032901"; # compatible with vscode 1.88.1
```
If you just rebase, you won't have to fix this here.

**Comment by superherointj** [comment]
@carlthome Can you rebase?

**Comment by superherointj** [comment]
@ofborg eval

**Comment by superherointj** [comment]
ping @carlthome

I'm waiting you to rebase to master to proceed.

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -2851,6 +2851,22 @@ let
         };
       };
 
+      ms-python.flake8 = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "flake8";
+          publisher = "ms-python";
+          version = "2023.6.0";
```
Comment (superherointj) [inline pkgs/applications/editors/vscode/extensions/default.nix:2858]:
```suggestion
          version = "2023.13.10681010";
```
I think there is a new version out there. Care to check?

Comment (carlthome) [reply pkgs/applications/editors/vscode/extensions/default.nix:2858]: Sorry, I'm realizing I don't have the bandwidth to have so many back and forth reviews on this small addition so closing this PR now.

Comment (superherointj) [reply pkgs/applications/editors/vscode/extensions/default.nix:2858]: https://github.com/microsoft/vscode-flake8/releases/tag/v2023.10.0

**Pull request closed without merging**